cg/services/orders/order_service/order_service.py

```diff
@@ -40,3 +40,11 @@ def create_order(self, order_data: OrderIn) -> OrderResponse:
 
     def set_delivery(self, order_id: int, delivered: bool) -> None:
         self.store.update_order_delivery(order_id=order_id, delivered=delivered)
+
+    def update_delivered(self, order_id: int, delivered_analyses: int) -> None:
+        order: Order = self.store.get_order_by_id(order_id)
+        case_count: int = len(order.cases)
+        if delivered_analyses >= case_count:
+            self.set_delivery(order_id=order_id, delivered=True)
+        else:
+            self.set_delivery(order_id=order_id, delivered=False)
```

Comment:
Is it worth to add some check to avoid setting this every time an analysis is marked as delivered? Might just clog the code with no added benefit though.